sql/2024/performance/web_vitals_by_technology.sql

```diff
@@ -0,0 +1,208 @@
+CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  SAFE_DIVIDE(good, (good + needs_improvement + poor)) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_POOR (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  SAFE_DIVIDE(poor, (good + needs_improvement + poor)) >= 0.25
+);
+
+CREATE TEMP FUNCTION IS_NI (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  NOT IS_GOOD(good, needs_improvement, poor) AND
+  NOT IS_POOR(good, needs_improvement, poor)
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+WITH base AS (
+  SELECT
+    date,
+    origin,
+    CONCAT(origin, '/') AS page,
+    CASE
+      WHEN device = 'phone' THEN 'mobile'
+      ELSE device
+    END AS client,
+
+    fast_fid,
+    avg_fid,
+    slow_fid,
+
+    fast_inp,
+    avg_inp,
+    slow_inp,
+
+    fast_lcp,
+    avg_lcp,
+    slow_lcp,
+
+    small_cls,
+    medium_cls,
+    large_cls,
+
+    fast_fcp,
+    avg_fcp,
+    slow_fcp,
+
+    fast_ttfb,
+    avg_ttfb,
+    slow_ttfb
+  FROM
+    `chrome-ux-report.materialized.device_summary`
+  WHERE
+    device IN ('desktop', 'phone') AND
+    date IN ('2024-06-01')
+),
+
+tech AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    category,
+    app AS technology,
+    url AS page
+  FROM
+    `httparchive.technologies.2024_06_01_*`
```

Comment:
This should be using `httparchive.all.pages` as it has better data and is being maintained going ahead.